pkg/blockbuilder/scheduler/metrics.go

```diff
@@ -0,0 +1,33 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package scheduler
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+)
+
+type schedulerMetrics struct {
+	monitorPartitionsDuration prometheus.Histogram
+	partitionStartOffsets     *prometheus.GaugeVec
+	partitionEndOffsets       *prometheus.GaugeVec
+}
+
+func newSchedulerMetrics(reg prometheus.Registerer) schedulerMetrics {
+	return schedulerMetrics{
+		monitorPartitionsDuration: promauto.With(reg).NewHistogram(prometheus.HistogramOpts{
+			Name: "cortex_blockbuilder_scheduler_monitor_partitions_duration_seconds",
+			Help: "Time spent monitoring partitions.",
+
+			NativeHistogramBucketFactor: 1.1,
+		}),
+		partitionStartOffsets: promauto.With(reg).NewGaugeVec(prometheus.GaugeOpts{
+			Name: "cortex_blockbuilder_scheduler_partition_start_offsets",
```

Comment:
Given it's 1 offset per partition, I would remove the final "s". Some comment applies to the end offset metric.

```suggestion
			Name: "cortex_blockbuilder_scheduler_partition_start_offset",
```